modules/formulize/include/functions.php

```diff
@@ -5320,6 +5334,130 @@ function generateHiddenElements($elements, $entry) {
     return $hiddenElements;
 }
 
+// Update for Ajax Save
+// Generate Form Instruction after a Save
+function genFormInstruction($info_continue, $fid, $entryId, $info_received_msg, $owner, $uid, $groups, $mid) {
+    $gperm_handler = & xoops_gethandler('groupperm');
+    if (!$owner) {
+        $owner = getEntryOwner($entry, $fid);
+    }
+    if (!$uid) {
+        $uid = $xoopsUser ? $xoopsUser->getVar("uid") : 0;
+    }
+    if (!$mid) {
+        $mid = getFormulizeModId();
+    }
+    if (!$groups) {
+        $groups = $xoopsUser ? $xoopsUser->getGroups() : array(0 => XOOPS_GROUP_ANONYMOUS);
+    }
+    $update_other_entries = $gperm_handler->checkRight("update_other_entries", $fid, $groups, $mid);
+    $update_own_entry = $gperm_handler->checkRight("update_own_entry", $fid, $groups, $mid);
+    $breakHTML = "";
+    if ($info_received_msg) {
+        $breakHTML .= _formulize_INFO_SAVED . "&nbsp;";
+    }
+    if ($info_continue == 1 AND (($owner == $uid AND $update_own_entry) OR $update_other_entries)) {
+        $breakHTML .= _formulize_INFO_CONTINUE1;
+    } elseif ($info_continue == 2) {
+        $breakHTML .= _formulize_INFO_CONTINUE2;
+    } elseif (!$entry AND ($gperm_handler->checkRight("add_own_entry", $fid, $groups, $mid) OR $gperm_handler->checkRight("add_proxy_entries", $fid, $groups, $mid))) {
+        $breakHTML .= _formulize_INFO_MAKENEW;
+    }
+    return $breakHTML;
+}
+
+// Generate Form Metadata
+function genFormMetaData($entryId, $fid, $member_handler) {
+    if (!$member_handler) {
+        $member_handler = & xoops_gethandler('member');
+    }
+    $form_meta = getMetaData($entryId, $member_handler, $fid);
+    $breakHTML = "<b>" . _formulize_FD_ABOUT . "</b><br>";
+    $breakHTML .= _formulize_FD_CREATED . $form_meta['created_by'] . " " . formulize_formatDateTime_duplicate_from_formdisplay_php($form_meta['created']) . "<br>" . _formulize_FD_MODIFIED . $form_meta['last_update_by'] . " " . formulize_formatDateTime_duplicate_from_formdisplay_php($form_meta['last_update']);
+    return $breakHTML;
+}
+
+// THIS FUNCTION FORMATS THE DATETIME INFO FOR DISPLAY CLEANLY AT THE TOP OF THE FORM
+// Old function is kept in formdisplay.php for historical reasons so nothing will break
```

Comment:
This comment makes it sound like we're purposely leaving duplicate code or functions around. We should just have one copy of the function. And it doesn't need a strange name like this. It's okay to move a function from formdisplay.php into functions.php. As long as the name is the same, the calling code will find the function. So verify that the two functions are the same, and once made equivalent, if any work is required to do so, then put the canonical copy in functions.php and make sure that everywhere that calls either function now calls the canonical copy in functions.php.

Reply:
removed the duplicate method in formdisplay.php